### DISCUSSION TITLE
Trouble with submitting dialog by keyboard

### DISCUSSION THREAD

**Discussion by ed2050** [zauberzeug/nicegui#2043] [Q&A] [answered]
### Question

I'm having trouble with submitting a dialog.  Here's what I want:
1. users presses to `/` to change filename
2. dialog pops up with a text box that has current filename
3. user edits text and hits `enter` to submit
4. server validates new filename and updates file

Screenshot:
<img width="517" alt="Screen Shot 2023-11-18 at 5 32 26 PM" src="https://github.com/zauberzeug/nicegui/assets/34918456/7e53dc1b-d3d5-496a-8344-78478e7ae451">

# Problem
I'm stuck on step 3.  I put a keyboard listener on my dialog with ui.keyboard.  Initially it gets triggered by keypresses.  Once text input has focus, nothing.  No more triggers, even on Enter.  The ui.input element swallows all keypresses.

If I refocus outside the input by clicking the dialog, then it works.  Keypresses go to my handler again.  BUT this approach breaks my workflow.  Everything has to be doable by keyboard without mouse.

How do I get my dialog keyboard listener to catch the Enter key in text input field?

## Code
```py
from nicegui import ui
from nicegui.events import KeyEventArguments

def dialog_key (e: KeyEventArguments, dialog, info) :

    DEBUG = true
    DEBUG and ui.notify (f'got a key!  { e.key }')

    if e.key == 'Enter' :

        DEBUG and ui.notify ('closing dialog')
        dialog.close ()
        # validate new filename info and rename 

def rename_dialog (filename) :

    info = {}
    info ['oldname'] = filename

    with ui.dialog () as dialog , ui.card ().classes ('dialog rename') :

        # static original name for user reference
        header ('Original')
        ui.label (filename)

        # input for new filename
        textin = ui.input (value = filename)
        textin.bind_value (info, 'newname')

        # set keyboard listener
        keyboard = ui.keyboard (on_key = lambda e : dialog_key (e, dialog, info), repeating = false)

    dialog.open ()

# another listener calls rename_dialog when user presses / 
```

**Comment by falkoschindler** [comment, marked as answer]
By default `ui.keyboard` ignores input elements like `ui.input`. But you can customize the ignore list:
```py
ui.keyboard(lambda e: ui.notify(f'{e.key} {e.action}'), ignore=[])
ui.input('Key events inside this input are not ignored')
```

**Reply by ed2050** [reply]
Thanks!  I was searching ui.input page, totally missed ignore flag on ui.keyboard.  Love when it's that easy.
